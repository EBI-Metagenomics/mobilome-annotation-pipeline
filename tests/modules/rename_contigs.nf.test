nextflow_process {

    name "Test Process RENAME"
    script "modules/rename_contigs.nf"
    process "RENAME"

    test("Should rename the sequences in the assembly") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = file("${projectDir}/tests/test_data/test.fasta")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
