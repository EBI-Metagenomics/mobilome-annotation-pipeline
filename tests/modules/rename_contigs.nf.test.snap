{
    "Should rename the sequences in the assembly": {
        "content": [
            {
                "0": [
                    "1kb_contigs.fasta:md5,d2d02a537b150e7fa0fe6c7aaedbf8b0"
                ],
                "1": [
                    "5kb_contigs.fasta:md5,70dd2c062dca462687219cf856bdd821"
                ],
                "2": [
                    "contigID.map:md5,5a013666031d5215b864cf3e6ae5c224"
                ],
                "contigs_1kb": [
                    "1kb_contigs.fasta:md5,d2d02a537b150e7fa0fe6c7aaedbf8b0"
                ],
                "contigs_5kb": [
                    "5kb_contigs.fasta:md5,70dd2c062dca462687219cf856bdd821"
                ],
                "map_file": [
                    "contigID.map:md5,5a013666031d5215b864cf3e6ae5c224"
                ]
            }
        ],
        "timestamp": "2023-11-28T21:41:12.632158"
    }
}